src/OpenTelemetry/Metrics/MeterProviderBuilderSdk.cs

```diff
@@ -79,6 +81,93 @@ internal MeterProviderBuilderSdk AddMetricReader(MetricReader metricReader)
             return this;
         }
 
+        internal MeterProviderBuilderSdk AddViewCallback(Func<Instrument, MetricStreamConfig> callback)
+        {
+            this.viewConfigs.Add(callback);
+            return this;
+        }
+
+        internal MeterProviderBuilderSdk AddView(string name, string meterName, string meterVersion, string instrumentName, InstrumentType instrumentType, string[] tagKeys, Aggregation aggregation, double[] histogramBounds)
+        {
+            if (string.IsNullOrWhiteSpace(meterVersion)
+                && string.IsNullOrWhiteSpace(meterName)
+                && string.IsNullOrWhiteSpace(instrumentName))
+            {
+                throw new ArgumentException("Atleast one instrument selection criteria should be provided.");
+            }
+
+            if (histogramBounds != null && aggregation != Aggregation.Histogram)
+            {
+                throw new ArgumentException("Histogram bounds are only applicable if Aggregation is Histogram.");
+            }
+
+            Func<Instrument, MetricStreamConfig> viewConfig = (instrument) =>
+            {
+                bool selectInstrument = false;
```

Comment:
`selectInstrument` will never become `true` with the `selectInstrument &= true` operations since it is `false` by default.

Reply:
good catch! will fix.